src/cbmc_starter_kit/update.py

```diff
@@ -0,0 +1,184 @@
+
+def migrate(cbmc_root, starter_kit_root):
+    logging.debug('Migrating CBMC starter kit')
+    templates = files_under_root(starter_kit_root/util.REPOSITORY_TEMPLATES)
+    logging.debug('Migrating CBMC starter kit: found templates: %s', templates)
+    symlinks = files_under_root(cbmc_root, symlinks_only=True)
+    logging.debug('Migrating CBMC starter kit: found symlinks: %s', symlinks)
+
+    cbmc_symlinks = [path for path in templates if path in symlinks]
+    for symlink in cbmc_symlinks:
+        cbmc_link = cbmc_root / symlink
+        cbmc_file = cbmc_link.resolve()
+        logging.warning('Copying: %s -> %s', cbmc_file, cbmc_link)
+        assert cbmc_file.exists()
+        assert cbmc_link.exists()
+        cbmc_link.unlink()
+        shutil.copy(cbmc_file, cbmc_link)
+
+def remove_negative_tests(cbmc_root):
+    logging.debug('Removing CBMC starter kit negative tests')
+    negative_tests = cbmc_root / 'negative_tests'
+    if negative_tests.is_dir():
+        logging.warning('Removing: %s', negative_tests)
+        shutil.rmtree(negative_tests)
+
+def update(cbmc_root):
+    logging.debug('Updating CBMC starter kit')
+    for path in [f'{util.PROOF_DIR}/{util.COMMON_MAKEFILE}', f'{util.PROOF_DIR}/{util.RUN_SCRIPT}']:
+        src = util.package_repository_template_root() / path
+        dst = cbmc_root / path
+        logging.warning('Copying: %s -> %s', src, dst)
+        version.copy_with_version(src, dst)
+
+def remove_starter_kit_submodule(cbmc_root, force=False):
+    logging.debug('Checking for starter kit submodule')
+    starter_kit = repository.starter_kit_root(repo=cbmc_root, abspath=False)
+    if not starter_kit:
+        logging.debug('Starter kit submodule does not exist')
+        return
+    if not force:
+        logging.warning('Use --force if you want to remove the starter kit submodule: %s',
+                        starter_kit)
+        return
+    for submodule in git.Repo(cbmc_root, search_parent_directories=True).submodules:
+        if submodule.path == str(starter_kit):
+            logging.warning('Removing: starter kit submodule: %s', starter_kit)
+            submodule.remove()
+            return
+    logging.error('This should be impossible: Failed to remove starter kit submodule: %s',
+                  starter_kit)
+
+################################################################
+
+def main():
+    """Migrate CBMC starter kit from submodule to pip package."""
+
+    args = parse_arguments()
+    logging.debug('args: %s', args)
+
+    args = validate_cbmc_root(args)
+    args = validate_starter_kit_root(args)
+
+    if not args.no_migrate:
+        migrate(args.cbmc_root, args.starter_kit_root)
+    if not args.no_test_removal:
+        remove_negative_tests(args.cbmc_root)
+    if not args.no_update:
+        update(args.cbmc_root)
+    if not args.no_starter_kit_removal:
+        remove_starter_kit_submodule(args.cbmc_root, args.force)
+
```

Comment:
Should the litani submodule, if one exists, be removed at exactly this point?

Reply:
I'd be happy to do that, but it is up to your team whether my tool should remove your work.

Reply:
I have made this an option.